catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -100,32 +102,39 @@ public static boolean enabled() {
    * <p>Caution should be exercised when executing code as system. Results from querying as system
    * are never returned to the client or cached. This means there is no risk of data leak.
    *
-   * @param framework the catalog framework, for creating system templates.
+   * @param catalogFramework the catalog framework, for creating system templates.
    * @param util for querying the catalog.
    * @param systemTemplates system templates loaded from config.
    */
   public static void bootstrap(
-      CatalogFramework framework, EndpointUtil util, List<Metacard> systemTemplates) {
-    Set<String> queryTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(QUERY_TEMPLATE_TAG)));
-    Set<String> resultTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(ATTRIBUTE_GROUP_TAG)));
+      CatalogFramework catalogFramework, EndpointUtil util, List<Metacard> systemTemplates) {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+    if (systemSubject == null) {
+      throw new SecurityException("Could not get systemSubject to initialize system templates");
+    }
 
-    List<Metacard> dedupedTemplateMetacards =
-        Stream.concat(
-                systemTemplates
-                    .stream()
-                    .filter(QueryTemplateMetacard::isQueryTemplateMetacard)
-                    .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
-                systemTemplates
-                    .stream()
-                    .filter(AttributeGroupMetacard::isAttributeGroupMetacard)
-                    .filter(metacard -> !resultTitles.contains(metacard.getTitle())))
-            .collect(Collectors.toList());
+    ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(4);
```

Comment:
❓ Why do we need 4 threads for the failsafe executor? Will a single thread work for this?

Reply:
single thread should work; 4 was just such a nice number

Reply:
I prefer 7 but i suppose that makes me pretty casual since its such a popular number

Reply:
You could use `Executors.newSingleThreadScheduledExecutor()`.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newSingleThreadScheduledExecutor--

Reply:
+1 for `Executors.newSingleThreadScheduledExecutor()`.

Reply:
done